src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/UseAsyncInAsync.cs

```diff
@@ -0,0 +1,265 @@
+                                                                              isDataflowRule: false);
+
+        public override ImmutableArray<DiagnosticDescriptor> SupportedDiagnostics => ImmutableArray.Create(Descriptor);
+
+        public override void Initialize(AnalysisContext context)
+        {
+            context.EnableConcurrentExecution();
+            context.ConfigureGeneratedCodeAnalysis(GeneratedCodeAnalysisFlags.None);
+            context.RegisterOperationBlockStartAction(context =>
+            {
+                context.RegisterOperationAction(AnalyzeInvocation, OperationKind.Invocation);
+
+                context.RegisterOperationAction(AnalyzePropertyGetter, OperationKind.PropertyReference);
+            });
+        }
+
+        internal static void AnalyzePropertyGetter(OperationAnalysisContext context)
+        {
+            if (IsInTaskReturningMethodOrDelegate(context))
+            {
+                InspectMemberAccess(context, CommonInterests.SyncBlockingProperties);
+            }
+        }
+
+        internal static void AnalyzeInvocation(OperationAnalysisContext context)
+        {
+            if (IsInTaskReturningMethodOrDelegate(context))
+            {
+                if (context.Operation is not IInvocationOperation)
+                {
+                    return;
+                }
+
+                if (InspectMemberAccess(context, CommonInterests.SyncBlockingMethods))
+                {
+                    // Don't return double-diagnostics.
+                    return;
+                }
+
+                // Also consider all method calls to check for Async-suffixed alternatives.
+                var semanticModel = context.Operation.SemanticModel;
+                SymbolInfo symbolInfo = semanticModel.GetSymbolInfo(context.Operation.Syntax, context.CancellationToken);
+                if (symbolInfo.Symbol is IMethodSymbol methodSymbol && !methodSymbol.Name.EndsWith(MandatoryAsyncSuffix, StringComparison.CurrentCulture) &&
+                    !methodSymbol.HasAsyncCompatibleReturnType())
+                {
+                    string asyncMethodName = methodSymbol.Name + MandatoryAsyncSuffix;
+                    ImmutableArray<ISymbol> symbols = semanticModel.LookupSymbols(
+                        context.Operation.Syntax.GetLocation().SourceSpan.Start,
+                        methodSymbol.ContainingType,
+                        asyncMethodName,
+                        includeReducedExtensionMethods: true);
+
+                    string containingMethodName = "";
+                    if (context.ContainingSymbol is IMethodSymbol parentMethod)
+                    {
+                        containingMethodName = parentMethod.Name;
+                    }
+
+                    if (context.Operation is not IInvocationOperation invOperation)
+                    {
+                        return;
```

Comment:
Isn't this check redundant? I think you already check that this is an invocation higher up in this method.